### DISCUSSION TITLE
ModelCheckpoint

### DISCUSSION THREAD

**Discussion by rjguedes8** [awslabs/gluonts#2190] [Q&A] [answered]
Is there a way to use a ModelCheckpoint to save the best model during training?
 - I'm using DeepVar Hierarchical and DeepAR

**Comment by lostella** [comment, marked as answer]
The first thing that comes to mind is a custom callback, see https://github.com/awslabs/gluon-ts/blob/08b6e2263ea5841c469a6624532133c81c45f1b1/src/gluonts/mx/trainer/callback.py#L33 for the callbacks API and some implementations, or the tutorial [here](https://ts.gluon.ai/stable/tutorials/advanced_topics/trainer_callbacks.html).

This will only allow you to serialize the model's network, and not the rest of the data loading or forecast generation pipeline, that make up a `Predictor` object. However, since you have the `Estimator` object, you can do:
- `network = <deserialize the network that you want>`
- `transform = estimator.create_transformation()`
- `predictor = estimator.create_predictor(transform, network)`

The above should work, although it could be exposed in a simpler way. The `create_transformation` and `create_predictor` are part of the `GluonEstimator` API, you can check their implementation for DeepAR (if you're interested in the details) [here](https://github.com/awslabs/gluon-ts/blob/08b6e2263ea5841c469a6624532133c81c45f1b1/src/gluonts/mx/model/deepar/_estimator.py#L301) and [here](https://github.com/awslabs/gluon-ts/blob/08b6e2263ea5841c469a6624532133c81c45f1b1/src/gluonts/mx/model/deepar/_estimator.py#L452)

**Reply by rjguedes8** [reply]
Thank you for the quick reply, I managed to create the Model Checkpoint and also another one to resume the training!
